### PR TITLE
Split run-script into multiple operations, closes #56

### LINKED ISSUE #56: split run-script into multiple operations
to better facilitate incremental undos. i think it makes sense for it to be three separate steps:

- `clone`, which takes `repos` and outputs `clones`
- `run-script`, takes `clones` and outputs `local-branches`
- `push`, takes `local-branches` and outputs `remote-branches`

(this does run the risk of introducing more concepts #55)

### PR DESCRIPTION
**Pull request by quarterto**
based on #76, merge that first

- [x] manual testing
- [x] add tests for the new operations

### CODE REVIEW

File: test/commands/run-script.test.js [hunk 1]
```diff
@@ -43,8 +35,8 @@ test('errors when `script` is empty', async () => {
 		runScript.handler({
 			workspace: 'hello'
 		}, {
-			repos: [{ owner: 'Financial-Times', name: 'next-search-page' }],
-		})
+				repos: [{ owner: 'Financial-Times', name: 'next-search-page' }],
```
Comment (quarterto) [inline test/commands/run-script.test.js]: u wot vs code why'd you format it like this

File: src/commands/run-script.js [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 
 const assert = require('assert');
 const fs = require('mz/fs');
 const path = require('path');
 const git = require('@financial-times/git');
-const rmfr = require('rmfr');
 
-const { workspacePath } = require('../lib/constants');
+const { GitProcess } = require('dugite');
+const constructDugiteExecArgs = require('@financial-times/git/src/helpers/construct-dugite-exec-args');
```
Comment (taktran) [inline src/commands/run-script.js]: `@financial-times/git` doesn't exist anymore right?

Comment (quarterto) [reply src/commands/run-script.js]: [?????](https://github.com/Financial-Times/tooling-helpers/tree/master/packages/git)

Comment (taktran) [reply src/commands/run-script.js]: Sorry, you are right, I was checking https://github.com/Financial-Times/git, forgot it was in https://github.com/Financial-Times/tooling-helpers/tree/master/packages/git

File: package-lock.json [hunk 3]
```diff
@@ -1,7613 +1,7626 @@
 {
```
Comment (quarterto) [inline package-lock.json]: looooooooooooool

**Pull request merged**